CMakeLists.txt

```diff
@@ -44,6 +44,9 @@ if(UNIX AND NOT APPLE)
         vendor/arm/pmu/pmu_profiler.cpp)
 endif()
 
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -fvisibility=hidden -fvisibility-inlines-hidden")
```

Comment:
This is not the recommended way of adding compiler flags.

If we don't have this properly done already within the project I would recommend we do this properly via https://cmake.org/cmake/help/v3.0/module/GenerateExportHeader.html. Which is the preferred way in CMake as well as more portable than adding compiler flags directly. This might need make version to be bumped up.